modules/c/nrt/source/Utils.c

```diff
@@ -295,30 +307,44 @@ NRTAPI(void) nrt_Utils_decimalToGeographic(double decimal, int *degrees,
     *minutes = (int) remainder;
     *seconds = fabs(remainder - (double) *minutes) * 60.0;
 
+    if (*degrees == 0)
+    {
+        if (*minutes == 0)
+        {
+            *seconds *= -1;
+        }
+        else
+        {
+            *minutes *= -1;
+        }
+    }
 }
 
 NRTAPI(double) nrt_Utils_geographicToDecimal(int degrees, int minutes,
                                              double seconds)
 {
     double decimal = fabs((double)degrees);
-    decimal += ((double) minutes / 60.0);
-    decimal += (seconds / 3600.0);
+    decimal += fabs(((double) minutes / 60.0));
+    decimal += fabs((seconds / 3600.0));
 
-    if (degrees < 0)
+    if ((degrees < 0) ||
```

Comment:
I think either you can...
1. Simplify this to just check `degrees < 0 || minutes < 0 || seconds < 0` (this is a valid check to do as long as the caller hasn't given you nonsense like positive degrees with negative minutes)
2. Add some sanity checking somewhere in the function to ensure that after the first non-zero DMS that all the remaining ones are positive.  Then you probably need to change the return type of the function to be able to provide an error.

I'd suggest just doing 1.